ddl/partition.go

```diff
@@ -1913,3 +1913,22 @@ func checkNoTimestampArgs(tbInfo *model.TableInfo, exprs ...ast.ExprNode) error
 	}
 	return nil
 }
+
+// sliceAnyOf returns true if any element in the slice matches the predict func.
+func sliceAnyOf(s interface{}, p func(int) bool) bool {
```

Comment:
em, how about put it to `util/slice`?

Reply:
sounds good to me, since ·br· is using these functions too, and Someone tells me the `br` will be merged into `tidb`, so make these function public will be better.